File: offset-manager/src/main/java/offsetmanager/domain/OffsetStorage.java
package offsetmanager.domain;

import offsetmanager.domain.file.FileKey;
import offsetmanager.domain.offset.OffsetRecord;

import java.util.Optional;

/**
 * High-level interface for offset storage operations <br>
 * OffsetStorage should store latest OffsetRecord identified by FileKey
 */
public interface OffsetStorage {
  Optional<OffsetRecord> find(FileKey key);
  void upsert(FileKey key, OffsetRecord record);
  void remove(FileKey key);
  void clear();
}

File: offset-manager/src/main/kotlin/offsetmanager/domain/OffsetStorage.kt
package offsetmanager.domain

import offsetmanager.domain.file.FileKey
import offsetmanager.domain.offset.OffsetRecord
import java.util.*

/**
 * High-level interface for offset storage operations <br></br>
 * OffsetStorage should store latest OffsetRecord identified by FileKey
 */
interface OffsetStorage {
  fun find(key: FileKey): Optional<OffsetRecord>
  fun upsert(key: FileKey, record: OffsetRecord)
  fun remove(key: FileKey)
  fun clear()
}
